crates/katana/executor/src/implementation/blockifier/utils.rs

```diff
@@ -74,14 +74,20 @@ pub fn transact<S: StateReader>(
     ) -> Result<(TransactionExecutionInfo, TxFeeInfo), ExecutionError> {
         let validate = !simulation_flags.skip_validate;
         let charge_fee = !simulation_flags.skip_fee_transfer;
+        // Blockifier doesn't provide a way to fully skip nonce check during the tx validation
+        // stage. The `nonce_check` flag in `tx.execute()` only 'relaxes' the check for
+        // nonce that is equal or higher than the current (expected) account nonce.
+        //
+        // Related commit on Blockifier: https://github.com/dojoengine/blockifier/commit/2410b6055453f247d48759f223c34b3fb5fa777
+        let nonce_check = !simulation_flags.skip_nonce_check;
```

Comment:
I would use the same vocabulary as the blockifier with this concept of `strict`. As we still check the nonce, but not using the `strict` equal policy.

WDYT? Minor detail, happy to merge like this. 👍

Reply:
Well, the simulation flags are already using an other nomenclature with `skip_nonce_check`, you can ignore my comment then. :)

Reply:
yeah the naming can be a bit confusing. hence why i put the comments there. i named it `skip_nonce_check` because `starknet_in_rust` had a flag to fully skip nonce check, so didnt rly put much thoughts into it. 

i might change the flag name later on to better reflect its behaviour inside blockifier.

Reply:
Will try to address this suggestion in later PR